zerver/lib/user_groups.py

```diff
@@ -58,6 +58,32 @@ def access_user_groups_as_potential_subgroups(
     return list(user_groups)
 
 
+def access_user_group_for_setting(
+    user_group_id: int,
+    user_profile: UserProfile,
+    *,
+    setting_name: str,
+    allow_only_system_groups: bool = False,
+    cannot_set_to_internet_group: bool = False,
+) -> UserGroup:
+    user_group = access_user_group_by_id(user_group_id, user_profile, for_read=True)
+
+    if allow_only_system_groups and not user_group.is_system_group:
+        raise JsonableError(
+            _("'{}' setting can only be set to system user groups.").format(setting_name)
+        )
+
+    if (
+        cannot_set_to_internet_group
+        and user_group.name == UserGroup.EVERYONE_ON_INTERNET_GROUP_NAME
+    ):
+        raise JsonableError(
+            _("'{}' setting cannot be set to '@role:internet' group.").format(setting_name)
```

Comment:
Not sure we want to mention this name here or description of the group like "Everyone on the internet".

Reply:
This is probably OK for now -- all these errors should be not user-facing anyway.

Oh, but for the translateable string -- can you make the group name a second parameter to `.format`? That'll avoid translators needing to do both this one and the next one with almost the same text.